src/content/tutorials/authorize-users-with-auth0/index.md

```diff
@@ -0,0 +1,995 @@
+
+### Configure an Auth0 application
+
+Every Auth0 account contains _applications_, which allow developers to create login/signup flows that are verified by Auth0. Integrating Auth0 with Workers requires creating an application in your Auth0 dashboard; if you’ve created an account for this tutorial, the “Default (Generic)” application provided by Auth0 will work; otherwise, create a new application with the type “Regular Web Application”.
+
+![Creating an Auth0 application](./media/creating-an-application.png)
+
+Inside of our application’s settings, the client ID and client secret are keys that we’ll provide to our Workers application to authenticate with Auth0. There are several settings and configuration options here, but relevant to this tutorial are the “Allowed Callback URLs” and “Allowed Web Origins” options. In the “Publish” section of this tutorial, we’ll fill in these values with the final deployed URL of our application.
+
+## Generate a new project
+
+Using wrangler’s `generate` command, we can begin building a new application using a Workers template. For this tutorial, we’ll heavily modify the default template for [Workers Sites](https://developers.cloudflare.com/workers/sites), which deploys a static HTML application:
+
+```sh
+---
+header: Generate a new project
+---
+$ wrangler generate --site my-auth-example
+```
+
+## Building an authorizer
+
+Before we begin implementing an “authorizer” in our application, which will verify that a user is logged in, it’s useful to understand how Auth0’s login flow works. The condensed version of this flow is below (and, if you’re interested, you can [read a longer writeup in Auth0’s docs](https://auth0.com/docs/flows/concepts/auth-code)):
+
+1. A user makes a request to the Workers application
+2. If they aren’t logged in, they’re redirected to the login page
+3. After logging in they’re redirected back to the Workers application, passing a login `code` as a query param
+4. The Workers application takes the login `code` parameter and exchanges it with Auth0 for authorization tokens
+
+In a traditional application that is attached to a database, the authorization tokens returned from Auth0 are often persisted in a database, allowing users to return to the application and continue to use it, without the need for re-authorization. With a Workers application, we have a quick and easy-to-use data storage solution that lives right next to our serverless application: Workers KV. Using Workers KV, we’ll store authorization tokens and tie them to a user using an authorization cookie.
+
+[![Auth0 Flow](./media/auth0-flow.png)](https://auth0.com/docs/flows/concepts/auth-code)
+
+<small>Auth0 Flow Diagram courtesy of <a href="https://auth0.com/docs/flows/concepts/auth-code">Auth0</a></small>
```

Comment:
Is this supposed to be a `<figcaption/>`? If so, maybe we should use that? Something like this perhaps:

```html
<figure>
<a href="https://auth0.com/docs/flows/concepts/auth-code">
<img src="./media/auth0-flow.png" alt="Auth0 flow">
</a>
<figcaption>Auth0 Flow Diagram courtesy of <a href="https://auth0.com/docs/flows/concepts/auth-code">Auth0</a></figcaption>
</figure>
```